### ISSUE TITLE
Question about the Hungarian Algorithm Implementation 

### ISSUE THREAD

**Issue by fedyhajali** [favyen/uns20#4]
Hi @uakfdotb,  
Below I quote what is reported in the paper.

"On intermediate frames, we apply the Hungarian method on M(0,k) to match detections in Dk with tracks, updating each track with the matched detection (if any)."

I would ask if get_recur_sel() function, together with tf.gather(), represents the Hungarian algorithm implementation.

https://github.com/favyen/uns20/blob/510833a221f891d638d21f4606fb88deae3a98dd/model.py#L366-L381
https://github.com/favyen/uns20/blob/510833a221f891d638d21f4606fb88deae3a98dd/model.py#L456-L466

**Comment by uakfdotb** [comment]
Yes, this is the part that matches detections with track prefixes during training. It is not the Hungarian algorithm but an approximation in this implementation since doing the bipartite matching is more complex and unnecessary.
